src/main/java/org/broadinstitute/dsm/model/Filter.java

```diff
@@ -281,15 +282,27 @@ private static boolean isNotEmpty(NameValue filter) {
     /**
      * Uses the appropriate date converter (if given) to write SQL that can
      * compare either exact dates or "in the day" dates.
+     * @param filter
+     * @param dbElement
+     * @param comparison how the values will be compared to one another
+     * @param arg the user-input field to compare
+     * @param useEndOfday if false, when parsing a date, the first millis of the day
+     *                    will be used.  if true, the last millis of the day will
+     *                    be used.
      */
-    private static String generateDateComparisonSql(Filter filter, DBElement dbElement, String comparison, Object arg) {
+    private static String generateDateComparisonSql(Filter filter, DBElement dbElement, String comparison, Object arg, boolean useEndOfday) {
         String column = filter.getColumnName(dbElement);
         SqlDateConverter dateConverter = null;
         if (dbElement != null) {
             dateConverter = dbElement.getDateConverter();
             Instant instant = null;
             try {
-                instant = LocalDate.parse(arg.toString(), DateTimeFormatter.ISO_LOCAL_DATE).atStartOfDay().toInstant(ZoneOffset.UTC);
+                LocalDate date = LocalDate.parse(arg.toString(), DateTimeFormatter.ISO_LOCAL_DATE);
+                if (useEndOfday) {
+                    instant = date.atTime(LocalTime.MAX).toInstant(ZoneOffset.UTC);
+                } else {
+                    instant = date.atStartOfDay().toInstant(ZoneOffset.UTC);
+                }
             } catch (DateTimeParseException e) {
                 // might be an epoch time in an older saved filter
                 instant = Instant.ofEpochMilli(Long.valueOf(arg.toString()));
```

Comment:
I think using `Long.parseLong()` method would be useful to avoid extra operation by using `valueOf`.